src/anyio/_backends/_asyncio.py

```diff
@@ -303,6 +301,12 @@ def cancel(self) -> None:
     def deadline(self) -> float:
         return self._deadline
 
+    @deadline.setter
+    def deadline(self, value: float) -> None:
+        self._deadline = float(value)
+        if self._active:
+            self._timeout()
```

Comment:
don't you need to cancel the old `self._timeout_handle`?

Reply:
Yes, good catch!